types/validation.go

```diff
@@ -58,7 +58,7 @@ func (h *SignedHeader) ValidateBasic() error {
 	}
 
 	// Handle Based Rollup case
-	if len(h.Validators.Validators) == 0 {
+	if h.Validators == nil || len(h.Validators.Validators) == 0 {
```

Comment:
can `h.Validators` ever be `nil`? i don't think so, because it is always initialized to empty ValidatorSet. If `nil` assumption holds then we have many places in the code base that does not check for `nil`.

Reply:
Yes, it can be `nil` when converting back `FromProto` here: https://github.com/rollkit/rollkit/blob/7e9ba94989f4a0ed60b191693b09b9468c6d2eff/types/serialization.go#L92

I think there might be a subtle difference here between them. The `Validators` field in the signed header itself CAN be nil (that is what we're checking for) but the `Validators` inside the `h.Validators` field CANNOT be nil. 

We're checking this in `ValidateBasic` for the `SignedHeader` which is called for every new `SignedHeader` we see, so I think this should suffice. If not, could you point out a place in the code base where it might not suffice?

Reply:
Should we instead, when converting back fromProto make sure that it is not nil but empty Validator instead ?

Reply:
documented in #876